### PR TITLE
feat(FN-2310): display payment link on payment details page

### PR DESCRIPTION
**Pull request by JoshBinns2000**
## Introduction :pencil2:
We want to render payments not at the `RECONCILED` or `READY_TO_KEY` status as links to the `edit-payment` page.

## Resolution :heavy_check_mark:
- Updates the template
- Updates the mapping function to retain the payment id (so that the correct link can be generated)
- Updates tests

### CODE REVIEW

File: trade-finance-manager-ui/templates/utilisation-reports/_macros/payment-details-table-row.njk
Comment (RLavender98) [inline trade-finance-manager-ui/templates/utilisation-reports/_macros/payment-details-table-row.njk:15]: I think I would prefer it if the logic of which to display as a link lived in the javascript, and then the view model had a field on paymentDetails that is the `editPaymentLink` which is either a string or undefined and we make it a link if it is defined and the userCanEdit here? That way we can use the status constant for the logic rather than hardcoding it here. What do you think, is there a reason for preferring this?

Comment (JoshBinns2000) [reply trade-finance-manager-ui/templates/utilisation-reports/_macros/payment-details-table-row.njk:15]: Both approaches are valid and I don't have a preference one way or the other. The reason for having the status check at this point was purely to match the way our existing templates perform this logic (we do a similar thing on the premium payments table for rendering links), so I think we should move both sets of logic to the view model if we prefer the `editPaymentLink?: string | undefined` approach to keep this consistent

**Review by RLavender98 — APPROVED** [review]
LGTM - Happy for you to merge once you also have approval from someone from UKEF

File: trade-finance-manager-ui/component-tests/utilisation-reports/_macros/payment-details-table-row.component-test.ts
Comment (abhi-markan) [inline trade-finance-manager-ui/component-tests/utilisation-reports/_macros/payment-details-table-row.component-test.ts:131]: Should we add assertions for `it.each([FEE_RECORD_STATUS.READY_TO_KEY, FEE_RECORD_STATUS.RECONCILED])(...` when `userCanEdit` is false?

File: trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.ts
Comment (abhi-markan) [inline trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.ts:240]: Can we rename `status` to `feeRecordPaymentGroupStatus`, so the return object is clean.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Only few minor comments to address please.

File: trade-finance-manager-ui/component-tests/utilisation-reports/_macros/payment-details-table-row.component-test.ts
Comment (JoshBinns2000) [reply trade-finance-manager-ui/component-tests/utilisation-reports/_macros/payment-details-table-row.component-test.ts:131]: We do - we have `it.each(Object.values(FEE_RECORD_STATUS))` which includes all the fee record statuses

File: trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.ts
Comment (JoshBinns2000) [reply trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.ts:240]: I have performed the rename in the destructuring of the type, but I would prefer not to rename the `FeeRecordPaymentGroup` `status` property itself. The reason to call it `feeRecordPaymentGroupStatus` in this context is that the individual payment does not have a "status", but the group it is in does

**Review by abhi-markan — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**
